docs/src/tutorials/software-management.md

```diff
@@ -1,67 +1,38 @@
+# Manage from Cumulocity the softwares on your devices
+This document describes how to manage the software modules that are installed on a thin-edge device from the 
+cloud using the **software management** feature of thin-edge.io.
 
-# Introduction
+> Note: As of now its supported only from Cumulocity cloud. Also, only supports **debian** based
+ distributions(Ex: RaspberryPi OS , Ubuntu, Debian), which use the **apt** package manager.
 
-This document describes how to install and use the **Software management** feature in thin-edge.io.
+Three components are required on your devices to enable software management:
 
-> Note: As of now the software management only supports **debian** based distros(Ex: Ubuntu, Debian),
- which use the **apt**.
+1. Software management mapper for Cumulocity cloud
+   Sm-mapper acts  as a proxy between the cloud and the device.
+   This translates the cloud specific message type into device specific type and vice-versa.(Example: Cumulocity smart-rest to/from thin-edge json)
+   The messages from cloud will be translated and forwarded to the `tedge_agent` and messages from `tedge_agent` will be translated and sent to cumulocity cloud.
+   You can find this process with the name `tedge_mapper sm-c8y` in `ps` once it starts.
 
+2. Software management agent
+   The thin-edge software management agent is the one that calls the plugins.
+   You can find this process with the name `tedge_agent` in `ps` once it starts.
 
+3. Software management plugin
+   Plugins are the interfaces that call the package manager (example: apt/apt-get) to do the software management operations (Install, Remove or update)
+   You can find them in /etc/tedge/sm-plugins.
+   As of now there is only one (apt) plugin is supported.
 
-## Download and Install software management packages on the device
 
-As a prequisite, install [tedge and tedge_mapper](../howto-guides/002_installation.md) if not installed already. 
+## Enable software management feature
+ 
+Find more information about [how to install and enable software management.](../../howto-guides/0012_install_and_enable_software_management.md)
+ 
+## Managing the device software on Cumulocity cloud
```

Comment:
```suggestion
## Managing the device software from Cumulocity cloud
```

Reply:
"on cumulocity" is correct here. Because we are managing the device software on the cloud. It's about adding/deleting a software to repo.

Reply:
Maybe the name repository is missing then?

Reply:
Okay. In that case, how about "Managing the device software **repository** on Cumulocity cloud"?